packages/devtools_extensions/lib/src/template/_simulated_devtools_environment/_simulated_devtools_environment.dart

```diff
@@ -74,73 +75,107 @@ class SimulatedDevToolsWrapperState extends State<SimulatedDevToolsWrapper>
-              children: [
-                Text(
-                  'Simulated DevTools Environment',
-                  style: theme.textTheme.titleMedium,
-                ),
-                const PaddedDivider(),
-                _VmServiceConnection(
-                  connected: connected,
-                  simController: simController,
-                ),
-                const SizedBox(height: denseSpacing),
-                _SimulatedApi(
-                  simController: simController,
-                  requiresRunningApplication: widget.requiresRunningApplication,
-                  connectedToApplication: connected,
-                ),
-                const PaddedDivider(),
-                Expanded(
-                  child: Column(
-                    crossAxisAlignment: CrossAxisAlignment.start,
-                    children: [
-                      Row(
-                        mainAxisAlignment: MainAxisAlignment.spaceBetween,
-                        children: [
-                          Text(
-                            'Logs:',
-                            style: theme.textTheme.titleMedium,
-                          ),
-                          DevToolsButton.iconOnly(
-                            icon: Icons.clear,
-                            outlined: false,
-                            tooltip: 'Clear logs',
-                            onPressed: () => simController.messageLogs.clear(),
-                          ),
-                        ],
-                      ),
-                      const PaddedDivider.thin(),
-                      Expanded(
-                        child: _LogMessages(simController: simController),
+    return LayoutBuilder(
+      builder: (context, constraints) {
+        final availableWidth = constraints.maxWidth;
+        const environmentPanelMinWidth =
+            _VmServiceConnection._totalControlsWidth + 2 * defaultSpacing;
+
+        final environmentPanelFraction =
+            environmentPanelMinWidth / availableWidth;
+        final childFraction = 1 - environmentPanelFraction;
+
+        return Split(
+          axis: Axis.horizontal,
+          initialFractions: [childFraction, environmentPanelFraction],
+          minSizes: const [
+            100.0,
+            0.0,
+          ],
+          children: [
+            OutlineDecoration.onlyRight(
+              child: Padding(
+                padding: const EdgeInsets.all(defaultSpacing),
+                child: widget.child,
+              ),
+            ),
+            LayoutBuilder(
+              builder: (context, environmentPanelConstraints) {
+                final availableEnvironmentPanelWidth =
+                    environmentPanelConstraints.maxWidth;
+                final environmentPanelWidth = math.max(
+                  environmentPanelMinWidth,
+                  availableEnvironmentPanelWidth,
+                );
+
+                return SingleChildScrollView(
```

Comment:
can you add a Scrollbar to this so that it is clear this section is horizontally scrollable?

Reply:
I've added this now. Please see [screen recording](https://github.com/flutter/devtools/pull/7062#issuecomment-1905315146).